File: LogisimPlus/LogisimPlus/draw/tools/ToolbarToolItem.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace draw.tools
{
    using LogisimPlus.Java;
    using ToolbarItem = draw.toolbar.ToolbarItem;

	public class ToolbarToolItem : ToolbarItem
	{
		private AbstractTool tool;
		private Icon icon;

		public ToolbarToolItem(AbstractTool tool)
		{
			this.tool = tool;
			this.icon = tool.Icon;
		}

		public virtual AbstractTool Tool
		{
			get
			{
				return tool;
			}
		}

		public virtual bool Selectable
		{
			get
			{
				return true;
			}
		}

		public virtual void paintIcon(JComponent destination, JGraphics g)
		{
			if (icon == null)
			{
				g.setColor(Color.FromArgb(255, 255, 128, 128));
				g.fillRect(4, 4, 8, 8);
				g.setColor(Color.Black);
				g.drawLine(4, 4, 12, 12);
				g.drawLine(4, 12, 12, 4);
				g.drawRect(4, 4, 8, 8);
			}
			else
			{
				icon.paintIcon(destination, g, 4, 4);
			}
		}

		public virtual string ToolTip
		{
			get
			{
				return tool.Description;
			}
		}

		public virtual Size getSize(object orientation)
		{
			if (icon == null)
			{
				return new Size(16, 16);
			}
			else
			{
				return new Size(icon.getIconWidth() + 8, icon.getIconHeight() + 8);
			}
		}
	}

}

File: logisim/app/src/main/java/draw/tools/ToolbarToolItem.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package draw.tools;

import java.awt.Color;
import java.awt.Component;
import java.awt.Dimension;
import java.awt.Graphics;

import javax.swing.Icon;

import draw.toolbar.ToolbarItem;

public class ToolbarToolItem implements ToolbarItem {
	private AbstractTool tool;
	private Icon icon;

	public ToolbarToolItem(AbstractTool tool) {
		this.tool = tool;
		icon = tool.getIcon();
	}

	public AbstractTool getTool() {
		return tool;
	}

	public boolean isSelectable() {
		return true;
	}

	public void paintIcon(Component destination, Graphics g) {
		if (icon != null) {
			icon.paintIcon(destination, g, 4, 4);
			return;
		}
		drawErrorIcon(g);
	}

	public static void drawErrorIcon(Graphics g) {
		g.setColor(new Color(255, 128, 128));
		g.fillRect(4, 4, 8, 8);
		g.setColor(Color.BLACK);
		g.drawLine(4, 4, 12, 12);
		g.drawLine(4, 12, 12, 4);
		g.drawRect(4, 4, 8, 8);
	}

	public String getToolTip() {
		return tool.getDescription();
	}

	public Dimension getDimension(Object orientation) {
		return icon == null
				? new Dimension(16, 16)
				: new Dimension(icon.getIconWidth() + 8, icon.getIconHeight() + 8);
	}
}
